### PR TITLE
fix(Breadcrumb): fix wrong overlay deprecation warning in Dropdown when use Breadcrumb with menu props

### LINKED ISSUE #40204: 官网 demo：Breadcrumb 使用 menu 参数，console 报警告 `overlay` is deprecated. Please use `menu` instead.
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/wp13k3?file=/demo.tsx:762-778)

### Steps to reproduce

打开 demo

### What is expected?

console 报警告

### What is actually happening?

console 不报警告

| Environment | Info |
| --- | --- |
| antd | 5.1.4 |
| React | 18 |
| System | Gentoo Linux 5.15.80 |
| Browser | Chrome 110.0.5481.24 |

---
原因是 `BreadcrumbItem` 组件里直接写了 `overlay`，导致 `props` 里有了 `overlay` 字段。
https://github.com/ant-design/ant-design/blob/52478c78d343fe68e0d57376e8613a9fb06fcd00/components/breadcrumb/BreadcrumbItem.tsx#L52

然后在 `Dropdown` 组件里用 `'overlay' in props` 来判断，所以报警告了。
https://github.com/ant-design/ant-design/blob/52478c78d343fe68e0d57376e8613a9fb06fcd00/components/dropdown/dropdown.tsx#L109-L113

### PR DESCRIPTION
**Pull request by candy4290**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/40204

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      fix wrong overlay deprecation warning in Dropdown when use Breadcrumb with menu props     |
| 🇨🇳 Chinese |      修复Breadcrumb组件使用menu属性，但是出现overlay deprecation警告的问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -105,9 +105,8 @@ const Dropdown: CompoundedComponent = (props) => {
         `\`${deprecatedName}\` is deprecated which will be removed in next major version, please use \`${newName}\` instead.`,
       );
     });
-
     warning(
-      !('overlay' in props),
+      !('overlay' in props && props.overlay !== undefined),
```
Comment (yoyo837) [inline components/dropdown/dropdown.tsx:109]: 后面的判断是覆盖了前面的判断的

**Comment by heiyu4585** [comment]
参考这个改下. https://github.com/ant-design/ant-design/issues/40204#issuecomment-1381318968

**Comment by candy4290** [comment]
> 参考这个改下. [#40204 (comment)](https://github.com/ant-design/ant-design/issues/40204#issuecomment-1381318968)

好的，我尝试下，确实感觉在BreadcrumbItem中来处理更合适

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 2]
```diff
@@ -48,8 +48,14 @@ const BreadcrumbItem: CompoundedComponent = (props) => {
...
   const renderBreadcrumbNode = (breadcrumbItem: React.ReactNode) => {
     if (menu || overlay) {
+      const mergeDropDownProps: DropdownProps = {
+        ...dropdownProps,
+      };
+      if ('overlay' in props) {
+        mergeDropDownProps.overlay = overlay;
+      }
       return (
-        <Dropdown menu={menu} overlay={overlay} placement="bottom" {...dropdownProps}>
```
Comment (afc163) [inline components/breadcrumb/BreadcrumbItem.tsx:52]: 看上去把 `menu={menu} overlay={overlay}` 删掉就行？

**Review by MadCcc — APPROVED** [review]

**Pull request merged**